### PR TITLE
feature/S4-108 - Generate DS01 PDF

### PR DESCRIPTION
**Pull request by peadarkelly**
**DO NOT MERGE** as environment variables are not yet available on Rebel1.

Dissolution API - https://github.com/companieshouse/dissolution-api/pull/24
Docker CHS Development - https://github.com/companieshouse/docker-chs-development/pull/23
PDF Template - TODO

* Example template can be found [here](https://s3.console.aws.amazon.com/s3/object/document-render-assets-registry/dissolution/templates/ds01.html?region=eu-west-1&tab=overview)
* Generated PDFs can be found [here](https://s3.console.aws.amazon.com/s3/buckets/dev-pdf-bucket/local/dissolution/?region=eu-west-1&tab=overview) for now until the new S3 bucket is created.
* Fixed logging config and changed implementation endpoint logs from debug -> info as logs were silent except for errors and debug already gives verbose endpoint logging.
* HTTP calls are being made using WebClient (as recommended [here](https://www.baeldung.com/rest-template#rest-template-is-deprecated)) and are tested using guidance from [here](https://www.baeldung.com/spring-mocking-webclient)
* Certificate is being saved in DB and details are being returned in DissolutionGetResponse.
* service/dissolution test files did not have the same folder structure as the impl files - fixed this.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/service/documentRender/DocumentRenderClient.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                .create(config.getDocumentRenderHost())
+                .post()
+                .uri(DOCUMENT_RENDER_STORE_ENDPOINT)
+                .header(HEADER_AUTHORIZATION, config.getApiKey())
+                .header(HEADER_ASSET_ID, HEADER_ASSET_ID_VALUE)
+                .header(HEADER_TEMPLATE_NAME, templateName)
+                .header(HEADER_ACCEPT, HEADER_ACCEPT_VALUE)
+                .header(HEADER_CONTENT_TYPE, HEADER_CONTENT_TYPE_VALUE)
+                .header(HEADER_LOCATION, location)
+                .body(Mono.just(asJsonString(data)), String.class)
```
Comment (peadarkelly) [inline src/main/java/uk/gov/companieshouse/service/documentRender/DocumentRenderClient.java:44]: I had to serialise the DissolutionCertificateData into a String as Spring was complaining that I was providing a Content-Type of `text/html` but trying to send JSON.

Although Spring is correct, the Document Render Service requires a Content-Type of `text/html` to be submitted in order for HTML -> PDF conversion to take place - see [here](https://github.com/companieshouse/document-render-service/blob/master/helpers/convert.go#L35)

File: src/main/java/uk/gov/companieshouse/model/dto/documentRender/DissolutionCertificateDirector.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    public String getOnBehalfName() {
+        return onBehalfName;
+    }
+
+    public void setOnBehalfName(String onBehalfName) {
+        this.onBehalfName = onBehalfName;
+    }
+
+    public Timestamp getApprovalDate() {
```
[the remark this answers is not in the document]
Comment (peadarkelly) [reply src/main/java/uk/gov/companieshouse/model/dto/documentRender/DissolutionCertificateDirector.java:27]: Originally used LocalDateTime but when this gets serialised it gets really messy., whereas Timestamp produces a raw value. This is the same as how we return dates in the DissolutionGetResponse

File: src/main/java/uk/gov/companieshouse/model/dto/documentRender/DissolutionCertificateData.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package uk.gov.companieshouse.model.dto.documentRender;
+
+import java.util.List;
+
+public class DissolutionCertificateData {
+
+    private String companyNumber;
```
[the remark this answers is not in the document]
Comment (peadarkelly) [reply src/main/java/uk/gov/companieshouse/model/dto/documentRender/DissolutionCertificateData.java:7]: @mateuszc-kainos and I agreed to expect camelCase property names in the PDF template. Happy to change if needed. What are you thoughts @mateuszc-kainos ?

[the remark this answers is not in the document]
Comment (mateuszc-kainos) [reply src/main/java/uk/gov/companieshouse/model/dto/documentRender/DissolutionCertificateData.java:7]: I would leave it as camel case personally. It's only used internally for render service.

[the remark this answers is not in the document]
Comment (matthewdoreilly) [reply src/main/java/uk/gov/companieshouse/model/dto/documentRender/DissolutionCertificateData.java:7]: 👍

File: src/main/java/uk/gov/companieshouse/mapper/DissolutionCertificateMapper.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        certificateDirector.setName(director.getName());
+        certificateDirector.setApprovalDate(getFormattedApprovalDate(director));
+        Optional.ofNullable(director.getOnBehalfName()).ifPresent(certificateDirector::setOnBehalfName);
+
+        return certificateDirector;
+    }
+
+    private String getFormattedApprovalDate(DissolutionDirector director) {
+        return director.getDirectorApproval().getDateTime().format(DateTimeFormatter.ofPattern(APPROVAL_DATE_FORMAT));
+    }
```
Comment (peadarkelly) [inline src/main/java/uk/gov/companieshouse/mapper/DissolutionCertificateMapper.java:60]: @mateuszc-kainos @matthewdoreilly Date formatting is now performed on the Dissolution API as we were unable to convert a Timestamp -> DD-MM-YYYY in the Golang template code.

**Pull request merged**